src/graphql/admin/mongoose_graphql_metric_admin_query.erl

```diff
@@ -5,179 +5,40 @@
 
 -ignore_xref([execute/4]).
 
--include("../mongoose_graphql_types.hrl").
 -include("mongoose_logger.hrl").
 
--import(mongoose_graphql_helper, [make_error/2, format_result/2]).
-
--type metric_result() :: term().
-
 execute(_Ctx, _Obj, <<"getMetrics">>, Args) ->
-    get_metrics(Args);
+    Name = get_name(Args),
+    mongoose_metrics_api:get_metrics(Name);
 execute(_Ctx, _Obj, <<"getMetricsAsDicts">>, Args) ->
-    get_metrics_as_dicts(Args);
+    Name = get_name(Args),
+    Keys = get_keys2(Args),
+    mongoose_metrics_api:get_metrics_as_dicts(Name, Keys);
 execute(_Ctx, _Obj, <<"getClusterMetricsAsDicts">>, Args) ->
-    get_cluster_metrics_as_dicts(Args).
-
--spec get_metrics(mongoose_graphql:args()) ->
-    {ok, [metric_result()]} | {error, resolver_error()}.
-get_metrics(Args) ->
-    Name = maps:get(<<"name">>, Args, []),
-    Values = exometer:get_values(prepare_name(Name)),
-    {ok, lists:map(fun make_metric_result/1, Values)}.
-
-get_metrics_as_dicts(Args) ->
-    Name = maps:get(<<"name">>, Args, []),
-    Keys = prepare_keys(maps:get(<<"keys">>, Args, null)),
-    Values = exometer:get_values(prepare_name(Name)),
-    {ok, [make_metric_dict_result(V, Keys) || V <- Values]}.
-
-get_cluster_metrics_as_dicts(Args) ->
-    Name = maps:get(<<"name">>, Args, []),
-    PrepName = prepare_name(Name),
-    Keys = prepare_keys(maps:get(<<"keys">>, Args, null)),
-    Nodes = prepare_nodes(maps:get(<<"nodes">>, Args, null)),
-    AllNodes = [node()|nodes()],
-    F = fun(Node) -> rpc:call(Node, exometer, get_values, [PrepName]) end,
-    FilteredNodes = filter_nodes(AllNodes, Nodes),
-    Results = mongoose_lib:pmap(F, FilteredNodes),
-    Zip = lists:zip(FilteredNodes, Results),
-    {ok, [make_node_result(Node, Result, Keys) || {Node, Result} <- Zip]}.
-
-make_node_result(Node, {ok, Values}, Keys) ->
-    {ok, #{<<"node">> => Node,
-           <<"result">> => [make_metric_dict_result(V, Keys) || V <- Values]}};
-make_node_result(Node, Other, _Keys) ->
-    ?LOG_ERROR(#{what => metric_get_failed,
-                 remote_node => Node, reason => Other}),
-    {error, <<"Failed to get metrics">>}.
+    Name = get_name(Args),
+    Keys = get_keys2(Args),
+    Nodes = get_nodes(Args),
+    mongoose_metrics_api:get_cluster_metrics_as_dicts(Name, Keys, Nodes).
 
-prepare_keys([]) ->
-    null;
-prepare_keys(null) ->
-    null;
-prepare_keys(Keys) ->
+get_keys2(Args) ->
```

Comment:
Why is not just `get_keys`?

Reply:
because get_keys is a bif. And there would be a warning.